### ISSUE TITLE
Resources 

### ISSUE THREAD

**Issue by mdcovarr** [adjacentlink/emane-tutorial#14]
Hello I was wondering if there are any resources to help me hook up traffic from a device or application to a NEM instance. Anything would help thanks

**Comment by sgalgano** [comment]
You will want to route application traffic to the vif which corresponds to the NEM.

Usually you are running a routing protocol on your nodes and configuring that protocol to use *emane0* as the over-the-air interface. The tutorial uses OLSR for routing and mgen for (application) traffic generation. If you have a more complex network architecture, you'll want to configure applications running on hosts using radio nodes as next hops.

**Comment by mdcovarr** [comment]
I thought I needed to push traffic to the platformendpoint which should then forward the traffic to the NEM instance. I currently do not have a complex network. I have 2 distributed NEM containers (one per platform). Both NEMs are configured to support RFPipe, and have transport="external". I am currently trying to pump some  UDP packets to the node-2 instance. However when I ssh into the node-2 container and do a 
`sudo tcpdump`
I don't see any packets being received. My setup very similar to the tutorials with OLSR for routing. My goal is to get some traffic from my localhost to a NEM container, but it seems like I am missing something in order to do that.

**Comment by sgalgano** [comment]
That is definitely possible.

You need to setup your routes (and/or routing protocols) so that the respective routers use their corresponding nodes (NEMs running in emulator instances in either containers, physical nodes or VMs) as next hops. You'll probably want to run a MANET routing protocol on your nodes running `emane` such that the [Virtual Transport](https://github.com/adjacentlink/emane/wiki/Virtual-Transport) *vifs* are configured as the interface the protocol uses for over-the-air messaging.

**Comment by mdcovarr** [comment]
Still not there for me yet.   If my physical interfaces of my emane computer (host) are enp02s20u4 (connected to cisco router 1) and enp02s20u1 (connected to cisco router 2) how do I get the data from enp02s20u4 into the emane simulation at all?   (Sorry for all the questions...)

**Comment by sgalgano** [comment]
You have two options: add a physical interface to each of your containers (assuming you are using LXC w/ emane) and then setup your routing configuration appropriately; or you can use the [Raw Transport](https://github.com/adjacentlink/emane/wiki/Raw-Transport) somewhat similar to [demo 7](https://github.com/adjacentlink/emane-tutorial/wiki/Demonstration-7), configure each of your NEMs to use *external* transports  and run `emaentransportd` on your host. The raw transport instances would be configured to use your physical interfaces and the routers would be connected as if they are on the same subnet.

**Comment by mdcovarr** [comment]
Hi Steven, thanks for the info; with it I was able to configure EMANE to allow external data to be passed through the OTA. I went ahead and added additional virtual ethernet pairs to the NEM containers. Additionally I configured the OLSR daemons to advertise the respective external networks; this handled configuring the next hops. Finally by bridging external interfaces to the respective veth interface on the host I was able get the behavior I desired. Thanks
